### PR TITLE
Add 5.1 test for present map type modifier

### PR DESCRIPTION
**Pull request by krishols**
Adapted from 5.0 test for close map type modifier

### CODE REVIEW

File: tests/5.1/target/test_target_map_with_present_modifier.c [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  } member; 
+
+  member.var = 1;
+  
+  for (i = 0; i < N; i++) { 
+    a[i] = i;
+    member.b[i] = i;
+  }
+
+#pragma omp target map (present, tofrom: scalar, a, member)
```
Comment (tob2) [inline tests/5.1/target/test_target_map_with_present_modifier.c:38]: This testcase has a bare **target** with **present** modifier. Thus, the following applies:

> If a **map** clause appears on a **target**, **target data**, **target enter data** or **target exit data** construct with a **present** _map-type-modifier_ then on entry to the region if the corresponding list item does not appear in the device data environment then an error occurs and the program terminates.

As in sollve_vv, there is no real infrastructure to test for should-fail tests, I assume the testcase should run. But in that case, the **target** construct must either be inside a **target data** region that maps the variables or the data must be mapped before with **target enter data**. 

(Or a combination of both: some variables one way, some the other way.)

* For a **target exit data**, the **present** modifier could also be used as that should work, given the variables have been mapped before.
* For the **target** construct, an **always** modifier would be required in addition, if the map-to value has been modified between the 'target (enter) data' and the 'target' — and when the result should be checked before the 'target exit data' or insie the 'target data' region, respectively.

Comment (nolanbaker31) [reply tests/5.1/target/test_target_map_with_present_modifier.c:38]: Might be useful to refer to 5.1/target_update/test_target_update_to_present.c

**Comment by nolanbaker31** [comment]
Closes #649

**Comment by krishols** [comment]
Passes with LLVM

**Review by spophale — APPROVED** [review]

**Review by mjcarr458 — APPROVED** [review]
looks good to me.

**Pull request merged**